projects/observability/src/shared/components/cartesian/d3/data/series/cartesian-column.ts

```diff
@@ -34,14 +34,18 @@ export class CartesianColumn<TData> extends CartesianSeries<TData> {
       );
 
       context.closePath();
-      context.strokeStyle = this.series.color;
-      context.fillStyle = this.series.color;
+      context.strokeStyle = this.getColorForDatum(datum);
+      context.fillStyle = this.getColorForDatum(datum);
       context.fill();
       context.stroke();
       context.resetTransform();
     });
   }
 
+  private getColorForDatum(datum: TData): string {
+    return this.series.getColor ? this.series.getColor(datum) : this.series.color;
```

Comment:
this can shorten with optional function access, but a broader question - wouldn't it make more sense to support specifying a color on the data object rather than a lookup function? that is: `{x, y, color}`

Reply:
it is possible, but the actual contents of the series is generic. Doing it that way would be much more intrusive and may have a broader impact across cartesians.

Reply:
updated to use optional function access.